kerykeion/charts/spreading.py

```
@@ -131,9 +158,38 @@ def spread_around_wheel(
+    # non-decreasing.
+    max_span = 360.0 - (seam_needed * shrink)
+    slack = max_span - sum(gaps_needed)
+    fitted_span = fitted[-1] - fitted[0]
+    if fitted_span > slack:
+        centre = (fitted[0] + fitted[-1]) / 2.0
+        # slack <= 0 means the gaps alone fill the budget: the crowd collapses
+        # onto the ramp and every gap lands exactly on its requirement.
+        squeeze = slack / fitted_span if slack > 0 else 0.0
+        fitted = [centre + (value - centre) * squeeze for value in fitted]
+
     placed = [value + offset for value, offset in zip(fitted, ramp)]
```

Ruff (0.16.2)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)